cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -408,7 +402,8 @@ void SparsePoseGraph::AddSubmapFromProto(const int trajectory_id,
   CHECK_EQ(optimized_submap_transforms_.at(trajectory_id).size(),
            submap_id.submap_index);
   optimized_submap_transforms_.at(trajectory_id)
-      .push_back(sparse_pose_graph::SubmapData{initial_pose_2d});
+      .emplace(optimized_submap_transforms_.at(trajectory_id).size(),
```

Comment:
RIght now this is the same value, but using `submap_id.submap_index` seems easier to read and more likely to be correct going forward, right?